packages/snaps-execution-environments/src/common/test-utils/endowments.ts

```diff
@@ -0,0 +1,126 @@
+// @ts-expect-error Walker has no types yet.
+import LavaTube from '@lavamoat/lavatube';
+import ObjectMultiplex from '@metamask/object-multiplex';
+import { StreamProvider } from '@metamask/providers';
+import { RequestArguments } from '@metamask/providers/dist/BaseProvider';
+import { SNAP_STREAM_NAMES } from '@metamask/snaps-utils';
+import { assert } from '@metamask/utils';
+import { ethErrors } from 'eth-rpc-errors';
+import { createIdRemapMiddleware } from 'json-rpc-engine';
+
+import { proxyStreamProvider, withTeardown } from '../utils';
+
+// These are fake types just to make this test work with the TypeScript
+/* eslint-disable @typescript-eslint/naming-convention */
+export type HardenedEndowmentSubject = {
+  __flag: unknown;
+  prototype: { __flag: unknown };
+};
+export type HardenedEndowmentInstance = {
+  __flag: unknown;
+  __proto__: { __flag: unknown };
+};
+/* eslint-enable @typescript-eslint/naming-convention */
+
+/**
+ * This function represents utility which is executed inside a compartment.
+ * It will try to change a subject or its prototype.
+ * Potential errors are caught and reported in an array returned.
+ *
+ * @param subject - Test subject (instance, object, function).
+ * @param factory - Factory that creates an instance using constructor function.
+ * @returns Array of error messages.
+ */
+export function testEndowmentHardening(
```

Comment:
(nitpicking)

I understand it feels right to put it here for reuse, but at the same time - the way we attempt to leak is important for understanding the test.  
Putting a copy of this into each test hurts the DRY patch in our hearts, I get it. But maybe it'd make sense to put just this function in a separate file where it'd be the least intimidating. Also, technically this function is a kind of a fixture.